modules/elasticsearch/elasticsearch.go

```diff
@@ -110,6 +100,63 @@ func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustom
 	return esContainer, nil
 }
 
+// certWriter is a helper that writes the details of a CA cert to options.
+type certWriter struct {
+	options  *Options
+	certPool *x509.CertPool
+}
+
+// Read reads the CA cert from the reader and appends it to the options.
+func (w *certWriter) Read(r io.Reader) error {
+	buf, err := io.ReadAll(r)
+	if err != nil {
+		return fmt.Errorf("read CA cert: %w", err)
+	}
+
+	w.options.CACert = buf
+	w.certPool.AppendCertsFromPEM(w.options.CACert)
+
+	return nil
+}
+
+// setWaitFor sets the req.WaitingFor strategy based on settings.
+func setWaitFor(options *Options, req *testcontainers.ContainerRequest) {
+	var strategies []wait.Strategy
+	if req.WaitingFor != nil {
+		// Custom waiting strategy, ensure we honour it.
+		strategies = append(strategies, req.WaitingFor)
+	}
+
+	waitHTTP := wait.ForHTTP("/").WithPort(defaultHTTPPort)
+	if sslRequired(req) {
+		waitHTTP = waitHTTP.WithTLS(true).WithAllowInsecure(true)
+		cw := &certWriter{
+			options:  options,
+			certPool: x509.NewCertPool(),
+		}
+
+		waitHTTP = waitHTTP.
+			WithTLS(true, &tls.Config{RootCAs: cw.certPool})
+		if options.Password != "" || options.Username != "" {
+			waitHTTP = waitHTTP.WithBasicAuth(options.Username, options.Password)
+		}
+		strategies = append(strategies, wait.ForFile(defaultCaCertPath).WithMatcher(cw.Read))
```

Comment:
BTW I enjoyed a lot understanding the behaviour of the wait.ForFile and this matcher represented by simply reading the file after copying it.

Really smart and super useful 👏